lib/OptimizationOptimisers/test/runtests.jl

```diff
@@ -32,6 +32,19 @@ using Lux, MLUtils, Random, ComponentArrays, Printf, MLDataDevices
     @test sol.stats.fevals == 1000
     @test sol.stats.gevals == 1000
 
+    @testset "epochs & maxiters" begin
+        optprob = SciMLBase.OptimizationFunction(
+            (u, data) -> sum(u) + sum(data), Optimization.AutoZygote())
+        prob = SciMLBase.OptimizationProblem(optprob, ones(2), ones(2))
+        @test_throws ArgumentError solve(prob, Optimisers.Adam())
+        @test_throws ArgumentError solve(
+            prob, Optimisers.Adam(), epochs = 2, maxiters = 2)
+        @test solve(prob, Optimisers.Adam(), epochs = 2)
```

Comment:
you can actually test for the expected error message and for the correct ones do this 

```
sol = solve(...)
@test sol.stats.iterations == 4
```